src/v/archival/archival_policy.cc

```diff
@@ -461,7 +461,7 @@ static ss::future<> get_file_range(
       upl);
     upl.content_length = upl.final_file_offset - upl.file_offset;
     vassert(
-      upl.content_length <= fsize,
+      upl.content_length <= segment->reader().file_size(),
```

Comment:
Can we add a comment with the PR letter here (situation in which this had led to incorrect assertion) for future readers of the code?